tests/Doctrine/Tests/ORM/Query/SelectSqlGenerationTest.php

```diff
@@ -84,6 +84,23 @@ public function assertInvalidSqlGeneration($dqlToBeTested, $expectedException, a
         $this->fail($sql);
     }
 
+    public function testDDC3697WithPartialLoad()
+    {
+        $this->assertSqlGeneration(
+            'SELECT c.id FROM \Doctrine\Tests\Models\Company\CompanyPerson c JOIN Doctrine\Tests\Models\Company\CompanyPerson r WHERE c.spouse = r AND r.id = 42',
+            'SELECT c0_.id AS id_0 FROM company_persons c0_ INNER JOIN company_persons c1_ WHERE c0_.spouse_id = c1_.id AND c1_.id = 42',
+            array(Query::HINT_FORCE_PARTIAL_LOAD => true)
+        );
+    }
+
+    public function testDDC3697WithoutPartialLoad()
```

Comment:
I would add (to the method name) something that indicates that `WHERE` and `WITH` both work